File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> out=new ArrayList<>();
        ArrayDeque<TreeNode> al=new ArrayDeque<>();
        if(root==null)
        return out;
        al.add(root);
        while(!al.isEmpty())
        {
            ArrayList<Integer> res=new ArrayList<>();
            int size=al.size();
            for(int i=0;i<size;i++)
            {
                TreeNode node=al.poll();
                res.add(node.val);
                if(node.left!=null)
                al.offer(node.left);
                if(node.right!=null)
                al.offer(node.right);
            }
            out.add(res);
        }
        return out;
    }
}

File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        al=[]
        res=[]
        if root==None:
            return []
        al.append(root)
        while len(al)>0:   
            res1=[] 
            size=len(al)
            while(size>0):
                node=al.pop(0)
                res1.append(node.val)
                if node.left!=None:
                    al.append(node.left)
                if node.right!=None:
                    al.append(node.right)
                size-=1    
            res.append(res1.copy())
        return res
